06-building-a-program-start-to-finish/writing-tests/output/spec.js

```diff
@@ -0,0 +1,147 @@
+    expect(typeof InvertedIndex.readFile).toBe('function');
+  });
+
+  it('Should contain the tokenize method', () => {
+    expect(typeof InvertedIndex.tokenize).toBe('function');
+  });
+
+  it('Should contain the validateFile method', () => {
+    expect(typeof InvertedIndex.validateFile).toBe('function');
+  });
+});
+describe('InvertedIndex createIndex', () => {
+  it('Should return true for creating Index', () => {
+    expect(invertedIndex.createIndex('books.json', myBook)).toBeTruthy();
+  });
+
+  it('Should return the numbers of books in a file', () => {
+    const numOfBooks = invertedIndex.booksIndex('books.json');
+    expect(numOfBooks.length).toEqual(6);
+  });
+});
+
+describe('InvertedIndex searchIndex', () => {
+  it('Should return `No file has been indexed yet', () => {
+    const search = () => {
+      invertedIndex.searchIndex('alice');
+    };
+    expect(search).toThrowError();
+  });
+
+  it('Should return an empty if not found', () => {
+    invertedIndex.searchIndex('alive', ['books.json']);
+    expect(invertedIndex.finalResult).toEqual(
+      { 'books.json': Object({ alive: [] }) }
+      );
+  });
+
+  it('Should return `please enter a keyword to search.`', () => {
+    const search = () => {
+      invertedIndex.searchIndex();
+    };
+    expect(search).toThrowError('please enter a keyword to search.');
+  });
+
+  it('Should return', () => {
+    expect(invertedIndex.getResult('alice', ['books.json']))
+      .toEqual({ alice: [0, 3] });
+  });
+
+  it('Should return', () => {
+    const validInput = () => {
+      invertedIndex.getResult('[[[[[[[[[[[]]]]]]]]]', ['books.json']);
+    };
+    expect(validInput).toThrow('Search for Aplhanumeric values only');
+  });
+
+  it('Should return an array of clean values', () => {
+    expect(InvertedIndex.cleanValues('How, are, you doing today'))
+      .toEqual(['how', 'are', 'you', 'doing', 'today']);
+  });
+});
+
+describe('InvertedIndex getIndex', () => {
+  it('Should return an object for getIndex method', () => {
+    expect(typeof getIndex).toBe('object');
+  });
+
+  it('Should return an alive as the first token', () => {
+    const alltoken = Object.keys(getIndex);
+    expect(alltoken[0]).toBe('alice');
+  });
+});
+
+describe('InvertedIndex readFile', () => {
+  it('Should return true for well formatted File', () => {
+    const book = JSON.stringify(myBook);
+    expect(InvertedIndex.validateFile(book, 'books.json')).toBeTruthy();
+  });
+
+  it('Should return an array for the JSON File', (done) => {
+    const readFile = InvertedIndex.readFile(jsonFile);
+    readFile.then((res) => {
+      expect(res[1][0].title).toBe('Alice in Wonderland');
+      done();
+    });
+  });
+  it('Should throw an error for empty books', () => {
+    const fileName = 'emptybook.json';
+    const checkEmptyBook = () => {
+      InvertedIndex.validateFile(emptyBook, fileName);
+    };
+    expect(checkEmptyBook).toThrowError();
+  });
```

Comment:
This test specification seems as though it should belong in your validateFile() Test Suite.